test/integration/acceptance/get_account_test.cpp

```diff
@@ -0,0 +1,317 @@
+   * @param user is account id for checking
+   * @param role is role for checking
+   * @return a lambda that verifies that query response contains created account
+   */
+  auto checkValidAccount(const std::string &domain,
+                         const std::string &user,
+                         const std::string &role) {
+    return [&](const proto::QueryResponse &response) {
+      ASSERT_NO_THROW({
+        const auto &resp = boost::apply_visitor(
+            framework::SpecifiedVisitor<interface::AccountResponse>(),
+            response.get());
+        ASSERT_EQ(resp.account().accountId(), user);
+        ASSERT_EQ(resp.account().domainId(), domain);
+        ASSERT_EQ(resp.roles().size(), 1);
+        ASSERT_EQ(resp.roles()[0], role);
+      }) << "Actual response: "
+         << response.toString();
+    };
+  }
+
+  /**
+   * @return a lambda that verifies that query response contains created account
+   */
+  auto checkValidAccount() {
+    return checkValidAccount(kDomain, kUserId, kNewRole);
+  }
+
+  /**
+   * @return a command for creating second user in the default domain
+   */
+  auto makeSecondUser() {
+    return complete(
+        createUserWithPerms(kUser2,
+                            kUser2Keypair.publicKey(),
+                            kRole2,
+                            {interface::permissions::Role::kSetQuorum}),
+        kAdminKeypair);
+  }
+
+  /**
+   * @return a command for creating second user in the new domain
+   */
+  auto makeSecondInterdomainUser() {
+    return complete(
+        baseTx()
+            .creatorAccountId(IntegrationTestFramework::kAdminId)
+            .createRole(kRole2, {interface::permissions::Role::kSetQuorum})
+            .createDomain(kNewDomain, kRole2)
+            .createAccount(kUser2, kNewDomain, kUser2Keypair.publicKey()),
+        kAdminKeypair);
+  }
+
+  const std::string kNewDomain = "newdom";
+  const std::string kNewRole = "rl";
+  const std::string kRole2 = "roletwo";
+  const std::string kUser2 = "usertwo";
+  const crypto::Keypair kUser2Keypair =
+      crypto::DefaultCryptoAlgorithmType::generateKeypair();
+  IntegrationTestFramework itf;
+};
+
+/**
+ * @given a user with all required permissions
+ * @when GetAccount is queried on the empty account name
+ * @then query is stateless invalid response
+ */
+TEST_F(GetAccount, EmptyAccount) {
+  prepareState().sendQuery(
+      makeQuery(""),
+      checkQueryErrorResponse<
+          shared_model::interface::StatelessFailedErrorResponse>());
+}
+
+/**
+ * @given a user with all required permissions
+ * @when GetAccount is queried on the user
+ * @then query is stateful invalid response
+ */
+TEST_F(GetAccount, NonexistentAccount) {
+  prepareState().sendQuery(
+      complete(baseQry().queryCounter(1).getAccount("inexistent@" + kDomain)),
+
+      checkQueryErrorResponse<
+          shared_model::interface::StatefulFailedErrorResponse>());
+}
+
+/**
+ * @given a user without any permission
```

Comment:
Why "without any permission"? that is not true. The user will have at least `Role::kSetQuorum` permission.

My initial suggestion was to state "without any *query* permission".